airbyte-webapp/src/components/connectorBuilder/types.ts

```diff
@@ -31,11 +34,95 @@ export interface BuilderStream {
     requestBody: Array<[string, string]>;
   };
 }
+function getInferredInputList(values: BuilderFormValues): BuilderFormInput[] {
+  if (values.global.authenticator.type === "ApiKeyAuthenticator") {
+    return [
+      {
+        key: "api_key",
+        required: true,
+        definition: {
+          type: "string",
+          title: "API Key",
+          airbyte_secret: true,
+        },
+      },
+    ];
+  }
+  if (values.global.authenticator.type === "BearerAuthenticator") {
+    return [
+      {
+        key: "api_key",
+        required: true,
+        definition: {
+          type: "string",
+          title: "API Key",
+          airbyte_secret: true,
+        },
+      },
+    ];
+  }
+  if (values.global.authenticator.type === "BasicHttpAuthenticator") {
+    return [
+      {
+        key: "username",
+        required: true,
+        definition: {
+          type: "string",
+          title: "Username",
+        },
+      },
+      {
+        key: "password",
+        required: true,
+        definition: {
+          type: "string",
+          title: "Password",
+          airbyte_secret: true,
+        },
+      },
+    ];
+  }
+  return [];
+}
+
+export function getInferredInputs(values: BuilderFormValues): BuilderFormInput[] {
+  const inferredInputs = getInferredInputList(values);
+  return inferredInputs.map((input) =>
+    values.inferredInputOverrides[input.key]
+      ? {
+          ...input,
+          definition: { ...input.definition, ...values.inferredInputOverrides[input.key] },
+        }
+      : input
+  );
+}
 
 export const builderFormValidationSchema = yup.object().shape({
   global: yup.object().shape({
     connectorName: yup.string().required("form.empty.error"),
     urlBase: yup.string().required("form.empty.error"),
+    authenticator: yup.object({
+      header: yup.mixed().when("type", {
+        is: "ApiKeyAuthenticator",
+        then: yup.string().required("form.empty.error"),
+        otherwise: (schema) => schema.strip(),
+      }),
+      api_token: yup.mixed().when("type", {
+        is: "ApiKeyAuthenticator",
```

Comment:
I think this should be:
```suggestion
        is: (val: string) => ["ApiKeyAuthenticator", "BearerAuthenticator"].includes(val),
```

Reply:
You are correct that this is wrong, but as we moved these into user inputs we don't need to validate them here at all as the user can't set them. Removed along with the username/password stuff.